plugins/lua/buildingplan/itemselection.lua

```diff
@@ -81,107 +81,147 @@ function ItemSelection:init()
+                },
             },
-            on_click=self:callback('submit'),
-            visible=function() return self.num_selected < self.quantity end,
         },
-        widgets.Label{
-            frame={r=0, w=15, t=0, h=3},
-            text_pen=BUILD_TEXT_PEN,
-            text_hpen=BUILD_TEXT_HPEN,
-            text={
-                '               ', NEWLINE,
-                '    Continue   ', NEWLINE,
-                '               ',
+    }
+
+    self:addviews{
+        widgets.Panel{
+            view_id='body',
+            frame={t=self.subviews.header.frame.h, b=4},
+            subviews={
+                widgets.EditField{
+                    view_id='search',
+                    frame={l=1, t=0},
+                    label_text='Search: ',
+                    on_char=function(ch) return ch:match('[%l -]') end,
+                },
+                widgets.CycleHotkeyLabel{
+                    frame={l=1, t=2},
+                    key='CUSTOM_SHIFT_R',
+                    label='Sort by:',
+                    options={
+                        {label='Recently used', value=sort_by_recency},
+                        {label='Name', value=sort_by_name},
+                        {label='Amount', value=sort_by_quantity},
+                    },
+                    on_change=self:callback('on_sort'),
+                },
+                widgets.Panel{
+                    frame={l=0, t=3, r=0, b=0},
+                    frame_style=gui.INTERIOR_FRAME,
+                    subviews={
+                        widgets.FilteredList{
+                            view_id='flist',
+                            frame={t=0, b=0},
+                            case_sensitive=false,
+                            choices=choices,
+                            icon_width=2,
+                            on_submit=self:callback('toggle_group'),
+                        },
+                    },
+                },
             },
-            on_click=self:callback('submit'),
-            visible=function() return self.num_selected >= self.quantity end,
         },
-        widgets.FilteredList{
-            view_id='flist',
-            frame={t=3, l=0, r=0, b=4},
-            case_sensitive=false,
-            choices=choices,
-            icon_width=2,
-            on_submit=self:callback('toggle_group'),
-            edit_on_char=function(ch) return ch:match('[%l -]') end,
-        },
-        widgets.CycleHotkeyLabel{
-            frame={l=0, b=2},
-            key='CUSTOM_SHIFT_R',
-            label='Sort by:',
-            options={
-                {label='Recently used', value=sort_by_recency},
-                {label='Name', value=sort_by_name},
-                {label='Amount', value=sort_by_quantity},
+        widgets.Panel{
+            view_id='footer',
+            frame={l=1, r=1, b=0, h=3},
+            subviews={
+                --[[
+                add an explanation for this terribleness
+                --]]
```

Comment:
terribleness meaning the overwritten hyphens? yeah, maybe we should add a "key_label" parameter that overrides the resolved key name? that would get rid of the need to hackily overwrite the text. Not in this PR, though.

can this comment be removed, though?

Reply:
sorry, the usage of the word "terribleness" was probably influenced by having enjoyed this video a little too much https://youtu.be/k238XpMMn38?t=61

i meant to come back and write a more descriptive comment about what is going on in the next section, (didnt do it immediately 'cause it really takes me _a lot_ of time to find a "good way" to explain what's happening, and i still had to fix a lot more things at that point) ;_; 
but then i forgot about it, had things irl distract me and so i committed in a hurry...

by "explain what's happening" i mean the whole "we are using key_sep to move the text out of the way so we can overlay other labels on it, because the original characters suck" thing.

the point is:
how do i put it? like, how do we explain what's going on in the next section, in a concise _and_ thorough manner?
(in the end, that comment was left there so that we could have this discussion, i guess :joy:)
or do i just not put any comments at all? which yes, could be a viable option as well...